Issue 31 add email api and email configurations

Viloetsisi opened the pull request:
## Change Summary
This PR implements the full backend email functionality for Issue #31.  
It introduces Django email configuration, a reusable email utility function, and updates the environment variable template.  
The integration was validated successfully by sending real test emails inside the DevContainer.

### Change Form

- [x] The pull request title has an issue number  
  *Includes: `issue-31-Add_email_API_and_email_configurations`*

- [x] The change works by "Smoke testing" or quick testing  
  *A real email was successfully sent using `send_simple_email()` inside `manage.py shell`, confirming Gmail App Password setup works.*

- [ ] The change has tests  
  *NA — No automated email tests added due to time constraints and because email sending typically requires mocking external SMTP servers.*

- [x] The change has documentation  
  *`.env.example` updated to include required email environment variables and developer instructions.*

## Other Information
- Merge conflicts with `settings.py` (due to new S3 configuration from `main`) were resolved cleanly.  
- Installed `django-storages[boto3]` to support new S3 backend introduced in main.  
- Adjusted DevContainer `.env` database host to ensure backend starts correctly after merging latest main.  
- Email sending was confirmed to work with Gmail App Passwords and correct SMTP configuration.


# Related issue

- Resolve #31

ErikaKK reviewed (commented):
I got a few questions, will talk tomorrow

ErikaKK commented on server/.env.example:
```diff
@@ -19,4 +19,25 @@ AWS_SECRET_ACCESS_KEY=xxxx
+EMAIL_HOST_PASSWORD=your_app_password_here
+
+EMAIL_USE_TLS=True
+DEFAULT_FROM_EMAIL="Bloom Notifications <your_email@example.com>"
```
This looks like optional, so pls remove this, bc the title of the subject can tell it's bloom email

Viloetsisi commented:
This update introduces a new `base_booking_email.html` template that standardises the layout across all booking-related emails (header, content area, footer). The base template now includes a properly styled black header bar and a centred Bloom logo sourced from Figma.

### Key Changes
- Added `base_booking_email.html` to centralise shared layout and styling.
- Updated `booking_confirmed.html` and `booking_cancelled.html` to extend the new base template.
- Implemented a Figma-accurate header with a black bar and white logo container.
- Added `bloom_logo.png` to `static/images` and wired it into templates through `bloom_logo_url`.
- Updated `email_utils.py` to inject `bloom_logo_url` automatically when rendering HTML emails.
- Improved consistency, maintainability, and branding alignment across all booking emails.

### Result
Both confirmation and cancellation emails now match the Bloom design system, use cleaner template structures, and are easier to maintain going forward.

<img width="980" height="806" alt="Screenshot 2025-12-11 211903" src="https://github.com/user-attachments/assets/0202c7b0-63d8-4b48-b24c-9cc6b34ec322" />

<img width="1089" height="737" alt="Screenshot 2025-12-11 211915" src="https://github.com/user-attachments/assets/6cefa96f-f33c-4708-87cf-cd8197dcdbff" />
